src/main/java/gregtech/api/metatileentity/multiblock/MultiblockControllerBase.java

```diff
@@ -394,10 +409,12 @@ public void onRemoval() {
         }
     }
 
-    @SuppressWarnings("unchecked")
     public <T> List<T> getAbilities(MultiblockAbility<T> ability) {
-        List<T> rawList = (List<T>) multiblockAbilities.getOrDefault(ability, Collections.emptyList());
-        return Collections.unmodifiableList(rawList);
+        return Collections.unmodifiableList(getAbilitiesModifiable(ability));
+    }
+
+    public <T> List<T> getAbilitiesModifiable(MultiblockAbility<T> ability) {
```

Comment:
This is never called anywhere except for `getAbilities()`. Nor is it overridden by any subclass. What is its intended use?

Reply:
originally it was used to add extra fluid tanks from item handlers, but that's done differently now